src/ansys/turbogrid/core/multi_blade_row/multi_blade_row.py

```diff
@@ -0,0 +1,1130 @@
+        # print(f"Blade rows to mesh: {self._blade_rows_to_mesh.keys()}")
+        # for blade_row in self._blade_rows_to_mesh:
+        #    print(f"Blade row {blade_row} blades: {self._blade_rows_to_mesh[blade_row]}")
+
+    def set_multi_process_count(self, multi_process_count: int):
+        """
+        Set number of process to be used in parallel for meshing the selected blade rows.
+
+        Parameters
+        ----------
+        multi_process_count:int
+            The number of processes to be used in parallel.
+        """
+        num_rows_to_process = len(self._blade_rows_to_mesh)
+        if num_rows_to_process == 0:
+            self._multi_process_count = 0
+            return
+        max_producers = min(cpu_count(), num_rows_to_process)
+        multi_process_count_clamped = min(max(0, multi_process_count), max_producers)
+        num_producers = min(num_rows_to_process, multi_process_count_clamped)
+        num_producers = num_producers if num_producers > 0 else max_producers
+        # print(f"{cpu_count()=}")
+        # print(f"{max_producers=}")
+        # print(f"{multi_process_count_clamped=}")
+        # print(f"{num_producers=}")
+        self._multi_process_count = num_producers
+
+    def set_blade_row_gsfs(self, assembly_gsf: float):
+        """
+        Set the global size factor to be used for all the blade rows.
+
+        Parameters
+        ----------
+        assembly_gsf:float
+            The Global Size Factor to be used for each blade row.
+            If not called the default size factor of 1 will be used.
+        """
+        self._blade_row_gsfs = self._get_blade_row_gsfs(assembly_gsf)
+        # print(f"blade_row_gsfs {self._blade_row_gsfs}")
+
+    def set_spanwise_counts(
+        self,
+        stator_spanwise_count: int,
+        rotor_spanwise_count: int,
+        rotor_blade_rows_blades: list = [],
+    ):
+        """
+        Set the spanwise count of mesh elements for the different blade rows.
+
+        Parameters
+        ----------
+        stator_spanwise_count:int
+            The element count in the spanwise direction for stator blade rows.
+        rotor_spanwise_count:int
+            The element count in the spanwise direction for rotor blade rows.
+        rotor_blade_rows:list, default: []
+            List of main blade from each rotor blade row. If not provided,
+            All even numbered rows will be taken as rotor with row numbering starting at 1.
+        """
+        self._blade_row_spanwise_counts = self._get_blade_row_spanwise_counts(
+            stator_spanwise_count, rotor_spanwise_count, rotor_blade_rows_blades
+        )
+
+    def set_blade_first_element_offsets(
+        self, assembly_f_el_offset: float, custom_blade_f_el_offsets: dict = None
+    ):
+        """
+        Set the boundary layer first element offset for the blades.
+
+        Parameters
+        ----------
+        assembly_f_el_offset:float
```

Comment:
"assembly_f_el_offset" is obscure. Can we just call it "assembly_offset"? (since "first element" is in the function name).
Similar for the second argument.

Reply:
Changed all "_f_el_offset" to "_bl_offset". Also changed the method name to: set_blade_boundary_layer_offsets. I guess this name will give a hint on what "bl" stands for.